### ISSUE TITLE
[java] UnusedLocalVariable: False positive for pattern variable in for-each without braces

### ISSUE THREAD

**Issue by FabioMarangonSMI** [pmd/pmd#6328]
**Affects PMD Version:** 7.19.0

**Rule:** [UnusedLocalVariable](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedlocalvariable)

**Description:** 
using an **instanceof** with a pattern variable inside a for-each without brackets gives a false positive

**Code Sample demonstrating the issue:**

```java
public void test() {
	Object[] array = {1, "a"};
	for(Object o : array)
		if(o instanceof String s)
			System.out.println(s);
	return;
}

```

**Expected outcome:**
Variable s should not be reported as unused. If i use the brackets on the for-each, the warning disappears.

PMD reports a violation at line "instanceof", but that's wrong. That's a false positive.

**Running PMD through:** *Maven*

With this code the warning is **not** reported:
```java
public void test() {
	Object[] array = {1, "a"};
	for(Object o : array) {
		if(o instanceof String s)
			System.out.println(s);
	}
	return;
}

```

**Comment by mdhamed238** [comment]
I am on it.
